### PR TITLE
[Migrillian] Make admission control laxer

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
Different logs use different x509 parsers, so when Migrillian populates a mirror
it might encounter a certificate/chain that it can't parse. This change makes
admission control in this case laxer: Migrillian only requires that the TLS
encoding of the entry complies with RFC6962.

### CODE REVIEW

File: trillian/migrillian/core/util.go [hunk 1]
```diff
@@ -59,27 +60,25 @@ func ValidateConfig(cfg *configpb.MigrationConfig) error {
...
-		cert = ct.ASN1Cert{Data: logEntry.X509Cert.Raw}
-	case logEntry.Precert != nil:
-		isPrecert = true
-		cert = logEntry.Precert.Submitted
-	default:
-		return nil, fmt.Errorf("entry at %d is neither cert nor pre-cert", index)
+	// Don't return on x509 parsing errors because we want to migrate this log
+	// entry as is. But log the error so that it can be flagged by monitoring.
+	if _, err = rle.ToLogEntry(); x509.IsFatal(err) {
+		glog.Errorf("%s: index=%d: x509 fatal error: %v", logPrefix, err)
```
Comment (daviddrysdale) [inline trillian/migrillian/core/util.go]: Wrong number of arguments (x2)

Comment (pavelkalinnikov) [reply trillian/migrillian/core/util.go]: Indeed.

File: trillian/migrillian/core/util.go [hunk 2]
```diff
@@ -59,27 +60,25 @@ func ValidateConfig(cfg *configpb.MigrationConfig) error {
 }
 
 func buildLogLeaf(logPrefix string, index int64, entry *ct.LeafEntry) (*trillian.LogLeaf, error) {
-	logEntry, err := ct.LogEntryFromLeaf(index, entry)
-	if x509.IsFatal(err) {
-		return nil, fmt.Errorf("failed to build LogEntry[%d]: %v", index, err)
+	rle, err := ct.RawLogEntryFromLeaf(entry, index)
```
Comment (daviddrysdale) [inline trillian/migrillian/core/util.go]: As it's only just gone in, can we also reverse the order of arguments in `RawLogEntryFromLeaf` so it's consistent with `LogEntryFromLeaf` please.

Comment (pavelkalinnikov) [reply trillian/migrillian/core/util.go]: Done.

**Pull request merged**